src/main/java/org/eclipse/yasson/internal/JsonbRiParser.java

```diff
@@ -271,7 +271,15 @@ public JsonObject getObject() {
 
     @Override
     public JsonValue getValue() {
-        return jsonParser.getValue();
+        final LevelContext currentLevel = level.peek();
+        switch (currentLevel.getLastEvent()) {
```

Comment:
Just a minor note we could use `switch (getLastEvent())`. Either way there is a potential for an NPE here if the `level` queue is empty.

Reply:
I just did the same it is in `skipJsonStructure` method. If you want I can check `isEmpty` first and just delegate to `jsonParser.getValue()` in that case.

Reply:
+1 that works too.

Reply:
It's done! Checking empty and possible null value in the event itself.